src/openrct2/platform/Android.cpp

```diff
@@ -62,4 +62,51 @@ bool platform_get_steam_path(utf8* outPath, size_t outSize)
     return false;
 }
 
+AndroidClassLoader::AndroidClassLoader()
+{
+    log_info("Obtaining JNI class loader");
+
+    // This is a workaround to be able to call JNI's ClassLoader from non-main
+    // thread, based on https://stackoverflow.com/a/16302771
+    // Apparently it's OK to use it from across different thread, but JNI
+    // only looks for ClassLoader in the _current_ thread and fails to find
+    // it when searched for from a native library's non-main thread.
+
+    // If we're here, SDL's JNI_OnLoad has already been called and set env
+    JNIEnv* env = (JNIEnv*)SDL_AndroidGetJNIEnv();
+
+    // First load some our random class
+    auto randomClass = env->FindClass("website/openrct2/MainActivity");
```

Comment:
Ah, I see. I think the comment could benefit from explaining *why* any arbitrary class can be chosen, then, rather than just saying that you're picking one at random; having read the code and the SO link I think I get why this works, but it feels like the right place to front-load a bit more explanation to me.